src/cli/commands/login.command.ts

```diff
@@ -38,18 +40,28 @@ export class LoginCommand {
             return Response.badRequest('Email address is invalid.');
         }
 
-        if ((password == null || password === '') && canInteract) {
-            const answer: inquirer.Answers = await inquirer.createPromptModule({ output: process.stderr })({
-                type: 'password',
-                name: 'password',
-                message: 'Master password:',
-            });
-            password = answer.password;
+        if (password == null || password === '') {
+            if (process.env.BW_PASSWORD) {
+                password = process.env.BW_PASSWORD;
+            } else if (canInteract) {
+                const answer: inquirer.Answers = await inquirer.createPromptModule({ output: process.stderr })({
+                    type: 'password',
+                    name: 'password',
+                    message: 'Master password:',
+                });
+                password = answer.password;
+            }
         }
         if (password == null || password === '') {
             return Response.badRequest('Master password is required.');
         }
-
+        if (password.startsWith('file:')) {
+            const filename = password.substring(5);
+            password = await NodeUtils.readFirstLine(filename);
+        }
+        if (password.startsWith('env:') && process.env[password.substring(4)]) {
+            password = process.env[password.substring(4)];
+        }
```

Comment:
What if someone's real password starts with "file:" or "env:"?

Reply:
I would think that is rare, also as a workaround you can set the real value into a file or environment variable.

Reply:
Could this be better handled with additional option flags?

Reply:
This is similar to how openssl handles passwords, so it's not totally unfamiliar to people that work with security tools. Maybe a clearest option would be to have a group of optional option flags like --password --password-file and --password-env, but it would change the current command usage and I was trying to make this backwards compatible.

Reply:
Adding new flags wouldn't be an issue with backwards compatibility. I like that option if it still solves the problem.

Reply:
Ok, I'll try and come up with a solution like that.